ogr/ogrsf_frmts/xodr/ogr_xodr.h

```diff
@@ -0,0 +1,261 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ ****************************************************************************/
+
+#pragma once
+#include "ogrsf_frmts.h"
+#include "ogr_api.h"
+#include <iostream>
+#include <OpenDriveMap.h>
+#include <pugixml/pugixml.hpp>
+#include <vector>
+
+struct RoadElements
+{
+    /* Map of road to its original OpenDRIVE ID for fast lookup. */
+    std::map<std::string, odr::Road> roads;
+    std::vector<odr::Line3D> referenceLines;
+
+    std::vector<odr::Lane> lanes;
+    std::vector<odr::LaneSection> laneSections;
+    std::vector<std::string> laneRoadIDs;
+    std::vector<odr::Mesh3D> laneMeshes;
+
+    std::vector<odr::Line3D> laneLinesInner;
+    std::vector<odr::Line3D> laneLinesOuter;
+
+    std::vector<odr::RoadMark> roadMarks;
+    std::vector<odr::Mesh3D> roadMarkMeshes;
+
+    std::vector<odr::RoadObject> roadObjects;
+    std::vector<odr::Mesh3D> roadObjectMeshes;
+
+    std::vector<odr::RoadSignal> roadSignals;
+    std::vector<odr::Mesh3D> roadSignalMeshes;
+};
+
+/*--------------------------------------------------------------------*/
+/*---------------------  Layer declarations  -------------------------*/
+/*--------------------------------------------------------------------*/
+
+class OGRXODRLayer : public OGRLayer
+{
+  private:
+    virtual void ResetReading() override;
+
+    virtual OGRFeatureDefn *GetLayerDefn() override
+    {
+        return featureDefn;
+    }
+
+    virtual int TestCapability(const char *) override
+    {
+        return FALSE;
+    }
+
+    /**
+     * Initializes XODR road elements and iterators.
+    */
+    void resetRoadElementIterators();
+
+  protected:
+    VSILFILE *file;
+    RoadElements roadElements;
+    OGRSpatialReference spatialRef;
+    bool dissolveTIN;
+    OGRFeatureDefn *featureDefn;
+
+    /* Unique feature ID which is automatically incremented for any new road feature creation. */
+    int nNextFID;
```

Comment:
Please use a default inline initialization or member variables, and use a m_ prefix. Applies to other similar places